src/app/routes/api/v3/forms/__tests__/public-forms.auth.routes.spec.ts

```diff
@@ -0,0 +1,272 @@
+import { ObjectId } from 'bson-ext'
+import { StatusCodes } from 'http-status-codes'
+import { err, errAsync } from 'neverthrow'
+import supertest, { Session } from 'supertest-session'
+
+import {
+  DatabaseError,
+  MissingFeatureError,
+} from 'src/app/modules/core/core.errors'
+import { AuthType, Status } from 'src/types'
+
+import { setupApp } from 'tests/integration/helpers/express-setup'
+import { buildCelebrateError } from 'tests/unit/backend/helpers/celebrate'
+import dbHandler from 'tests/unit/backend/helpers/jest-db'
+import { jsonParseStringify } from 'tests/unit/backend/helpers/serialize-data'
+
+import * as FormService from '../../../../../modules/form/form.service'
+import { MyInfoFactory } from '../../../../../modules/myinfo/myinfo.factory'
+import { CreateRedirectUrlError } from '../../../../../modules/spcp/spcp.errors'
+import { SpcpFactory } from '../../../../../modules/spcp/spcp.factory'
+import { PublicFormsRouter } from '../public-forms.routes'
+
+const app = setupApp('/forms', PublicFormsRouter)
+
+describe('public-form.auth.routes', () => {
+  let request: Session
+
+  beforeAll(async () => await dbHandler.connect())
+  beforeEach(async () => {
+    request = supertest(app)
+  })
+  afterEach(async () => {
+    await dbHandler.clearDatabase()
+    jest.restoreAllMocks()
+  })
+  afterAll(async () => await dbHandler.closeDatabase())
+  describe('POST /forms/:formId/auth/redirect', () => {
+    it('should return 200 with the redirect URL when the form is valid and has authType SP', async () => {
+      // Arrange
+      const { form } = await dbHandler.insertEncryptForm({
+        formOptions: {
+          authType: AuthType.SP,
+          status: Status.Public,
+          esrvcId: new ObjectId().toHexString(),
+        },
+      })
+
+      // Act
+      const response = await request
+        .get(`/forms/${form._id}/auth/redirect`)
+        .query({ isPersistentLogin: false })
+
+      // Assert
+      expect(response.status).toEqual(StatusCodes.OK)
+      expect(response.body).toMatchObject({
+        redirectURL: expect.any(String),
```

Comment:
do we know the shape of the redirectURL and how it differs when isPersistentLogin is true vs false? We should test that the return string fits a substring of some valid string etc.

Reply:
i could check with anty/alwyn on this and make the check stronger. cos atm this is actually handled by an external library that's written by them HAHA

Reply:
the redirectURL is created using an external library and for `authType = SP | CP`, the target has the form `(formId, {isPersistentLogin})` but not too sure for `MyInfo`. didn't wanna do that cos i thought external library supposed to be a black box @_@

i can clarify with anty/alwyn (cos they wrote the library) and update this!

Reply:
since this is an integration test, we should assert the string returned by the third party does not change (in the syntax/style) at least

Reply:
ah ok, good point! will do so (and will clarify for `MyInfo`)

Reply:
read through `MyInfo` implementation, the implementation is [here](https://github.com/opengovsg/myinfo-gov-client/blob/c6d7d24b5a7813631ea12005b3446641d8261bec/src/MyInfoGovClient.class.ts#L172) and i'll update it accordingly!